boot/bootutil/src/loader.c

```diff
@@ -1583,6 +1586,33 @@ context_boot_go(struct boot_loader_state *state, struct boot_rsp *rsp)
         /* Set the previously determined swap type */
         bs.swap_type = BOOT_SWAP_TYPE(state);
 
+#if defined(MCUBOOT_OVERWRITE_ONLY) && defined(MCUBOOT_DOWNGRADE_PREVENTION)
+        /* Check if version of secondary is sufficient */
+        rc = boot_is_version_sufficient(
+                &boot_img_hdr(state, BOOT_PRIMARY_SLOT)->ih_ver,
+                &boot_img_hdr(state, BOOT_SECONDARY_SLOT)->ih_ver);
+        if (rc != 0) {
+                const struct flash_area *fap;
+
+                rc = flash_area_open(
```

Comment:
This open is leaked (but see below, it won't really work anyway).

Reply:
I removed the entire "update the boot state" logic. AFAICT it is only used when resuming an interrupted swap operation (and this only works without swapping), so it is not needed.